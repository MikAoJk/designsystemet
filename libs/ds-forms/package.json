{
  "name": "@skatteetaten/ds-forms",
  "groupId": "no.skatteetaten.aurora",
  "version": "0.4.0",
  "publishConfig": {
    "registry": "https://nexus.sits.no/repository/npm-internal/"
  },
  "peerDependencies": {
    "i18next": "^22",
    "react": "^18",
    "react-i18next": "^12"
  }
}
