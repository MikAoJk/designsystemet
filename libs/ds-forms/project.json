{
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/ds-forms/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "build": {
      "executor": "@nrwl/web:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/ds-forms",
        "tsConfig": "libs/ds-forms/tsconfig.lib.json",
        "project": "libs/ds-forms/package.json",
        "entryFile": "libs/ds-forms/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "@nrwl/react/plugins/bundle-rollup",
        "compiler": "babel",
        "assets": [
          {
            "glob": "libs/ds-forms/README.md",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/ds-forms/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/libs/ds-forms"],
      "options": {
        "jestConfig": "libs/ds-forms/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "stylelint": {
      "executor": "nx-stylelint:lint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": [
          "libs/ds-forms/**/*.css",
          "libs/ds-forms/**/*.scss"
        ]
      }
    },
    "tsc": {
      "executor": "./tools/executors/tsCheck:tsCheck",
      "options": {
        "tsConfig": ["tsconfig.lib.json", "tsconfig.spec.json"]
      }
    },
    "set-version": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm version {args.packageVersion} --no-git-tag",
        "cwd": "libs/ds-forms"
      }
    },
    "publish": {
      "executor": "nx:run-commands",
      "options": {
        "command": "npm publish {args.switches}",
        "cwd": "dist/libs/ds-forms"
      }
    }
  }
}
