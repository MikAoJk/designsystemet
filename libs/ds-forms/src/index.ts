export * from './Checkbox/Checkbox';
export * from './Checkbox/Checkbox.types';
export * from './CheckboxGroup/CheckboxGroup';
export * from './CheckboxGroup/CheckboxGroup.types';
export * from './ErrorMessage/ErrorMessage';
export * from './ErrorMessage/ErrorMessage.types';
export * from './Radio/Radio';
export * from './Radio/Radio.types';
export * from './RadioGroup/RadioGroup';
export * from './RadioGroup/RadioGroup.types';
