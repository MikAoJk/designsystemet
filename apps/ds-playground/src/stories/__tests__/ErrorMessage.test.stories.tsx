import { ErrorMessage } from '@skatteetaten/ds-forms';
import { expect } from '@storybook/jest';
import { ComponentStory, ComponentMeta } from '@storybook/react';
import { within } from '@storybook/testing-library';

export default {
  component: ErrorMessage,
  title: 'Tester/ErrorMessage',
  argTypes: {
    // Baseprops
    key: { table: { disable: true } },
    ref: { table: { disable: true } },
    className: { table: { disable: true } },
    id: { table: { disable: true } },
    lang: { table: { disable: true } },
    'data-testid': { table: { disable: true } },
    // Props
    children: {
      table: { disable: true },
      control: 'text',
    },
    showError: {
      table: { disable: true },
    },
  },
} as ComponentMeta<typeof ErrorMessage>;

const defaultText = 'Feilmelding';

const defaultArgs = {
  children: defaultText,
};

const Template: ComponentStory<typeof ErrorMessage> = (args) => (
  <div data-test-block>
    <ErrorMessage {...args} />
  </div>
);

// Når ErrorMessage har en ref, så får dom elementet ref forwarded
export const WithRef = Template.bind({});
WithRef.storyName = 'With Ref (FA1)';
WithRef.args = {
  ...defaultArgs,
  ref: (instance: HTMLDivElement | null): void => {
    if (instance) {
      instance.id = 'dummyIdForwardedFromRef';
    }
  },
};
WithRef.argTypes = {
  ref: { table: { disable: false } },
};
WithRef.parameters = {
  imageSnapshot: { disable: true },
};
WithRef.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const container = canvas.getAllByRole('generic')[1];
  await expect(container).toHaveAttribute('id', 'dummyIdForwardedFromRef');
};

// Når ErrorMessage har en id, så har containeren id'en satt
// Når ErrorMessage har en custom className, så har containeren className satt og custom stil vises
// Når ErrorMessage har en lang, så har containeren lang satt
// Når ErrorMessage har en data-testid, så har containeren data-testid satt
export const WithAttributes = Template.bind({});
WithAttributes.storyName = 'With Attributes (FA2-5, B1)';
WithAttributes.args = {
  ...defaultArgs,
  id: 'htmlId',
  className: 'dummyClassname',
  lang: 'nb',
  'data-testid': '123ID',
  showError: true,
};
WithAttributes.argTypes = {
  id: { table: { disable: false } },
  className: { table: { disable: false } },
  lang: { table: { disable: false } },
  'data-testid': { table: { disable: false } },
};
WithAttributes.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const container = canvas.getAllByRole('generic')[1];
  await expect(container).toHaveAttribute('id', 'htmlId');
  await expect(container).toHaveClass('dummyClassname');
  await expect(container).toHaveAttribute('lang', 'nb');
  await expect(container).toHaveAttribute('data-testid', '123ID');
};

// Når ErrorMessage instansieres, får den riktige default-verdier og p-elementet finnes ikke
export const Defaults = Template.bind({});
Defaults.storyName = 'Default (A1, A2)';
Defaults.args = {
  ...defaultArgs,
};
Defaults.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const container = canvas.getAllByRole('generic')[1];
  await expect(container).toBeInTheDocument();
  await expect(container).toHaveAttribute('aria-live', 'assertive');
  await expect(container).toHaveAttribute('aria-atomic');
  // eslint-disable-next-line testing-library/no-node-access
  const textNode = container.querySelector('p');
  await expect(textNode).not.toBeInTheDocument();
};

// Når ErrorMessage har children, finnes et p-element
export const WithPElement = Template.bind({});
WithPElement.storyName = 'With <p>-element (B1)';
WithPElement.args = {
  ...defaultArgs,
  showError: true,
};
WithPElement.argTypes = {
  showError: {
    table: { disable: false },
  },
};
WithPElement.play = async ({ canvasElement }): Promise<void> => {
  const canvas = within(canvasElement);
  const textNode = canvas.getByText(defaultText);
  await expect(textNode.tagName).toBe('P');
};

const TemplateWithInput: ComponentStory<typeof ErrorMessage> = (args) => (
  <div className={'errorMessageContainer'} data-test-block>
    {/* TODO FRONT-1279 erstattes med TextField når den er ferdig utviklet */}
    <label htmlFor={'label1'} className={'block'}>
      {'Label'}
    </label>
    <input id={'label1'} type={'text'} />
    <ErrorMessage {...args} />
  </div>
);

// Når ErrorMessage har en tekst over flere linjer, så aligner ikonet med første linje
export const WithLongText = TemplateWithInput.bind({});
WithLongText.storyName = 'With Long Text (A3)';
WithLongText.args = {
  ...defaultArgs,
  showError: true,
  children:
    'Dette blir en veldig lang feilmelding for å teste om du oppfører seg om den skal.',
};
WithLongText.argTypes = {
  children: {
    table: { disable: false },
  },
};

// Når ErrorMessage har en lang tekst uten breaking space, så skal det brekke over flere linjer
export const WithLongTextAndBreaking = TemplateWithInput.bind({});
WithLongTextAndBreaking.storyName = 'With Long Text And Breaking';
WithLongTextAndBreaking.args = {
  ...defaultArgs,
  showError: true,
  children:
    'Detteblirenveldiglangfeilmeldingforåtesteomduoppførersegomdenskal.',
};
WithLongTextAndBreaking.argTypes = {
  children: {
    table: { disable: false },
  },
};
