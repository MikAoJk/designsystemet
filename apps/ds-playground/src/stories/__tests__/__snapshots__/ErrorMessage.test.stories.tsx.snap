// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tester/ErrorMessage Defaults play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       aria-atomic="true"
  >
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithAttributes play-test 1`] = `
<div data-test-block="true">
  <div id="htmlId"
       lang="nb"
       data-testid="123ID"
       aria-live="assertive"
       aria-atomic="true"
  >
    <div class="ErrorMessage_errorMessage__508oW dummyClassname">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo ErrorMessage_errorMessageIcon__2N-Qa"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
        </path>
      </svg>
      <p class="ErrorMessage_errorMessageText__e27TU">
        Feilmelding
      </p>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithLongText smoke-test 1`] = `
<div data-test-block="true">
  <div class="errorMessageContainer">
    <div class>
      <label for="textFieldTextboxId-:r3:"
             class="TextField_label__K0isy  undefined"
      >
        <span class>
          Label
        </span>
      </label>
      <input id="textFieldTextboxId-:r3:"
             class="TextField_textbox__qbZlk    undefined"
             aria-describedby="errorId"
             value
      >
      <div id="textFieldErrorId-:r2:"
           aria-live="assertive"
           aria-atomic="true"
      >
      </div>
    </div>
    <div id="errorId"
         aria-live="assertive"
         aria-atomic="true"
    >
      <div class="ErrorMessage_errorMessage__508oW ">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo ErrorMessage_errorMessageIcon__2N-Qa"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
        <p class="ErrorMessage_errorMessageText__e27TU">
          Dette blir en veldig lang feilmelding for å teste om du oppfører seg om den skal.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithLongTextAndBreaking smoke-test 1`] = `
<div data-test-block="true">
  <div class="errorMessageContainer">
    <div class>
      <label for="textFieldTextboxId-:r6:"
             class="TextField_label__K0isy  undefined"
      >
        <span class>
          Label
        </span>
      </label>
      <input id="textFieldTextboxId-:r6:"
             class="TextField_textbox__qbZlk    undefined"
             aria-describedby="errorId"
             value
      >
      <div id="textFieldErrorId-:r5:"
           aria-live="assertive"
           aria-atomic="true"
      >
      </div>
    </div>
    <div aria-live="assertive"
         aria-atomic="true"
    >
      <div class="ErrorMessage_errorMessage__508oW ">
        <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo ErrorMessage_errorMessageIcon__2N-Qa"
             viewbox="0 0 24 24"
             focusable="false"
             aria-hidden="true"
             role="img"
        >
          <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
          </path>
        </svg>
        <p class="ErrorMessage_errorMessageText__e27TU">
          Detteblirenveldiglangfeilmeldingforåtesteomduoppførersegomdenskal.
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithPElement play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       aria-atomic="true"
  >
    <div class="ErrorMessage_errorMessage__508oW ">
      <svg class="Icon_systemIcon__1p1Xl Icon_systemIcon_small__Yzxmo ErrorMessage_errorMessageIcon__2N-Qa"
           viewbox="0 0 24 24"
           focusable="false"
           aria-hidden="true"
           role="img"
      >
        <path d="M13 13h-2V7h2v6Zm0 4h-2v-2h2v2ZM12 2C6.477 2 2 6.477 2 12A10 10 0 1 0 12 2Z">
        </path>
      </svg>
      <p class="ErrorMessage_errorMessageText__e27TU">
        Feilmelding
      </p>
    </div>
  </div>
</div>
`;

exports[`Tester/ErrorMessage WithRef play-test 1`] = `
<div data-test-block="true">
  <div aria-live="assertive"
       aria-atomic="true"
       id="dummyIdForwardedFromRef"
  >
  </div>
</div>
`;
